unified-codes/apps/data-service/src/app/AuthorisationService.ts

```diff
@@ -0,0 +1,27 @@
+import User from './User';
+import IdentityProvider from './IdentityProvider';
+
+export interface IAuthorisationConfig {
+  baseUrl: string;
+}
+
+export class AuthorisationError extends Error {
+  constructor(message: string) {
+    super(message);
+  }
+}
+
+export class AuthorisationService {
+  private provider: IdentityProvider;
+
+  constructor(provider: IdentityProvider) {
+    this.provider = provider;
+  }
+
+  // TODO: add permissions, e.g. authorise(user: User, permission: Permission)
```

Comment:
Might need to have a discussion how to implement the permission logic? Maybe a `Permission` class and/or enum for roles?